contracts/GeneratedExperience.cdc

```diff
@@ -0,0 +1,351 @@
+			Type<MetadataViews.NFTCollectionDisplay>(),
+			Type<MetadataViews.Medias>(),
+			Type<MetadataViews.Rarity>(),
+			Type<FindPack.PackRevealData>()
+			]
+		}
+
+		pub fun resolveView(_ view: Type): AnyStruct? {
+
+			let collection = GeneratedExperience.collectionInfo[self.info.season]!
+			let imageFile = MetadataViews.IPFSFile( cid: self.info.thumbnailHash, path: nil)
+
+			switch view {
+
+			case Type<FindPack.PackRevealData>():
+				let data : {String : String} = {
+					"nftImage" : imageFile.uri() ,
+					"nftName" : self.info.name,
+					"packType" : "GeneratedExperience"
+				}
+				return FindPack.PackRevealData(data)
+
+			case Type<MetadataViews.Display>():
+				return MetadataViews.Display(
+					name: self.info.name,
+					description: self.info.description,
+					thumbnail: MetadataViews.IPFSFile(
+						cid: self.info.thumbnailHash, path: nil
+					)
+				)
+			case Type<MetadataViews.Editions>():
+				// We do not show season here unless there are more than 1 collectionInfo (that is indexed by season)
+				let editionName = "genereatedexperience"
+				let editionInfo = MetadataViews.Edition(name: editionName, number: self.info.edition, max: self.info.maxEdition)
+				let editionList: [MetadataViews.Edition] = [editionInfo]
+				return MetadataViews.Editions(
+					editionList
+				)
+			case Type<MetadataViews.Royalties>():
+				return MetadataViews.Royalties(collection.royalties)
+
+			case Type<MetadataViews.ExternalURL>():
+				if self.owner != nil {
+					return MetadataViews.ExternalURL("https://find.xyz/".concat(self.owner!.address.toString()).concat("/collection/main/generatedExperience/").concat(self.id.toString()))
+				}
+				return MetadataViews.ExternalURL("https://find.xyz/")
+
+			case Type<MetadataViews.NFTCollectionData>():
+				return MetadataViews.NFTCollectionData(
+					storagePath: GeneratedExperience.CollectionStoragePath,
+					publicPath: GeneratedExperience.CollectionPublicPath,
+					providerPath: GeneratedExperience.CollectionPrivatePath,
+					publicCollection: Type<&GeneratedExperience.Collection{NonFungibleToken.CollectionPublic,NonFungibleToken.Receiver,MetadataViews.ResolverCollection}>(),
+					publicLinkedType: Type<&GeneratedExperience.Collection{NonFungibleToken.CollectionPublic,NonFungibleToken.Receiver,MetadataViews.ResolverCollection}>(),
+					providerLinkedType: Type<&GeneratedExperience.Collection{NonFungibleToken.CollectionPublic,NonFungibleToken.Provider,MetadataViews.ResolverCollection}>(),
+					createEmptyCollectionFunction: (fun (): @NonFungibleToken.Collection {
+						return <-GeneratedExperience.createEmptyCollection()
+					})
+				)
+			case Type<MetadataViews.NFTCollectionDisplay>():
+
+				var square = MetadataViews.Media(
+					file: MetadataViews.IPFSFile(
+						cid: collection.squareImage,
+						path: nil
+					),
+					mediaType: "image/png"
+				)
+
+				var banner = MetadataViews.Media(
+					file: MetadataViews.IPFSFile(
+						cid: collection.bannerImage,
```

Comment:
I am not sure we can make this an ipfs file here can we? or we have to upload to ipfs in the job?